frontend/src/context/useHomeLocationQuery.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useLocationContext } from './useCurrentLocation';
+
+
+const useHomeLocationQuery = () => {
+  const { setCurrentLocation, coordinates } = useLocationContext()
+  const [requestLocation, setRequestLocation] = useState(false)
+  const { push } = useHistory();
+
+  const handleGeoLocation = useCallback((e) => {
+    e.preventDefault();
+    setCurrentLocation();
+    setRequestLocation(true);
+  },[setCurrentLocation])
+
+  useEffect(() => {
+    (requestLocation && coordinates.length) && push('/stores');
+  }, [coordinates, requestLocation, push])
+
+  return useMemo(() => [handleGeoLocation], [handleGeoLocation]);
```

Comment:
the second array is for the dependencies, you are saying the function depends on itself here. Also why are you returning an Array instead of the function? And why are you not using useCallback instead

Reply:
oh! yeah, It's better to return the function instead of additional `useMemo`.